pkg/util/pubsub/http_handler.go

```diff
@@ -72,47 +70,33 @@ func (h *HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	logger = logger.WithField("channel", channelName)
 
 	go func() {
-		redisClient := h.RedisPool.Get()
-		psc := redisClient.Subscribe(rootCtx, channelName)
+		msgChan, cancel, err := h.RedisHub.Subscribe(channelName)
+		if err != nil {
+			errChan <- err
+			return
+		}
 
 		defer func() {
-			psc.Close()
+			cancel()
 			logger.Info("redis goroutine is tearing down")
 		}()
 
 		for {
 			select {
 			case <-doneChan:
 				return
-			default:
-				// Even though ReceiveTimeout takes context,
-				// cancelling the context does not immediately make ReceiveTimeout return error.
-				// Therefore, we have to use a small enough timeout to ensure pubsub is closed within a reasonble delay.
-				iface, err := psc.ReceiveTimeout(rootCtx, RedisReadTimeout)
-				if os.IsTimeout(err) {
-					continue
+			case n, ok := <-msgChan:
+				if !ok {
+					return
 				}
+
+				writeCtx, cancel := context.WithTimeout(rootCtx, WebsocketWriteTimeout)
+				defer cancel()
+				err := wsConn.Write(writeCtx, websocket.MessageText, []byte(n.Payload))
```

Comment:
I think think may cause unresponsive pubsub mechanism when client is malicious:
Since `msgChan` (created in `Hub.Subscribe`) is un-buffered, the spawned message dispatch goroutine would block if the previous message is still sending (blocked at `wsConn.Write`).

Reply:
We have a reverse proxy in front of us (traefik), so I guess the write should finish pretty fast?

Reply:
Even the buffers of reverse proxy can be filled up if client is really slow (intentionally), and I don't think the proxies would drop the connection once it gets full? With some amount of (malicious) clients, this case may be hit frequently.